virtcontainers/sandbox.go

```diff
@@ -959,6 +959,12 @@ func (s *Sandbox) startVM() (err error) {
 
 	s.Logger().Info("Starting VM")
 
+	defer func() {
```

Comment:
In such a change, then if the error is triggered before the hypervisor startSandbox, would this cleanup tigger any other issues?  Is it better to do another cleanup in hypervisor.startSandbox() function?

Reply:
I agree

1. The defer should go after setup the network setup, in case that  `if err := s.network.Run(s.networkNS.NetNsPath, func() error {` fail the hypervisor is not started so the defer need to be a few lines after just before call `startSandbox()`

2.  I think that `startSandbox` should kill the hypervisor process in case of something goes wrong and not let the caller clean up if something fails. I see that actually it is killing the virtiofs process we should do something similar inside the function.